File: modules/expo-text-cursor/ios/ExpoTextCursorModule.swift
import ExpoModulesCore

public class ExpoTextCursorModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ExpoTextCursor")

    Function("getCharacterIndex") { (text: String, x: Double, y: Double, fontSize: Double, lineHeight: Double, containerWidth: Double) -> Int in
      return TextCursorCalculator.getCharacterIndex(
        text: text,
        x: CGFloat(x),
        y: CGFloat(y),
        fontSize: CGFloat(fontSize),
        lineHeight: CGFloat(lineHeight),
        containerWidth: CGFloat(containerWidth)
      )
    }
  }
}

File: modules/expo-text-cursor/src/ExpoTextCursorModule.ts
import { NativeModule, requireNativeModule } from "expo-modules-core";

declare class ExpoTextCursorModuleType extends NativeModule {
  getCharacterIndex(
    text: string,
    x: number,
    y: number,
    fontSize: number,
    lineHeight: number,
    containerWidth: number
  ): number;
}

export default requireNativeModule<ExpoTextCursorModuleType>("ExpoTextCursor");
